tools/ci_build/github/android/build_aar_package.py

```diff
@@ -94,6 +95,26 @@ def _build_aar(args):
     base_build_command = [sys.executable, BUILD_PY] + build_settings["build_params"] + ["--config=" + build_config]
     header_files_path = ""
 
+    if qnn_android_build:
+        qnn_home = args.qnn_path
+        sdk_file = os.path.join(qnn_home, "sdk.yaml")
+        qnn_sdk_version = None
+        with open(sdk_file) as f:
+            for line in f:
+                if line.strip().startswith("version:"):
+                    # yaml file has simple key: value format with version as key
+                    qnn_sdk_version = line.split(":", 1)[1].strip()
+                    break
+
+        # Note: The QNN package version does not follow Semantic Versioning (SemVer) format.
+        # only use major.minor.patch version for qnn sdk version and truncate the build_id info if any
+        # yaml file typically has version like 2.26.0
+        if qnn_sdk_version:
+            qnn_sdk_version = ".".join(qnn_sdk_version.split(".")[:3])
+            base_build_command += ["--qnn_home=" + qnn_home]
+        else:
+            print("QNN SDK version not found in sdk.yaml file.")
```

Comment:
maybe we should error out here or check for `qnn_sdk_version` being `None` below.